src/main.rs

```diff
@@ -1,3 +1,246 @@
+        //Draw the current state of the grid
+        draw(&matrix, &is_grid_showing);
+        //DRAW text
+        draw_text(status, 5.0, 30.0, 50.0, DARKBLUE);
+        draw_text(
+            "Cells can be edited on pause",
+            5.0,
+            screen_height() - 50.0,
+            25.0,
+            DARKBLUE,
+        );
+        draw_text(
+            "Press <space> to start/pause the game",
+            5.0,
+            screen_height() - 30.0,
+            25.0,
+            DARKBLUE,
+        );
+        draw_text(
+            "Press <G> to toggle grid on/off",
+            5.0,
+            screen_height() - 10.0,
+            25.0,
+            DARKBLUE,
+        );
+        //Do transitions
+        if is_running {
+            //Calculate neighbour_grid and transitions
+            let neighbour_grid = calculate_neighbour_amount_matrix(&matrix);
+            transition(&mut matrix, &neighbour_grid);
+        }
+        std::thread::sleep(std::time::Duration::from_millis(1));
+        next_frame().await
+    }
+}
+//Calculate transitions
+fn transition(matrix: &mut [Vec<i32>], neighbours: &[Vec<i32>]) {
+    for (row_index, row) in neighbours.iter().enumerate() {
+        for (col_index, col) in row.iter().enumerate() {
+            if matrix[row_index][col_index] == 1 {
+                match *col {
+                    0..=1 => matrix[row_index][col_index] = 0,
+                    2..=3 => matrix[row_index][col_index] = 1,
+                    _ => matrix[row_index][col_index] = 0,
+                }
+            } else if *col == 3 {
+                matrix[row_index][col_index] = 1;
+            }
+        }
+    }
+}
+//Process clicks
+fn process_clicks(matrix: &mut [Vec<i32>]) {
+    let height_scale = screen_height() / (matrix.len() as f32);
+    let width_scale = screen_width() / (matrix[0].len() as f32);
+    let (column_pixel, row_pixel) = mouse_position();
+    let row_index = (row_pixel / height_scale).floor() as usize;
+    let column_index = (column_pixel / width_scale).floor() as usize;
+    if matrix[row_index][column_index] == 1 {
+        matrix[row_index][column_index] = 0;
+    } else {
+        matrix[row_index][column_index] = 1;
+    }
+}
+//Calculate neighbours
+fn calculate_neighbour_amount_matrix(matrix: &[Vec<i32>]) -> Vec<Vec<i32>> {
+    let mut neighbour_amount_matrix = vec![vec![0; matrix[0].len()]; matrix.len()];
+    //Iterate through the rows of the grid
+    for (row_index, row) in matrix.iter().enumerate() {
+        //Iterate through the columns of the row
+        for (column_index, _) in row.iter().enumerate() {
+            //Calculate the amounts of neighbours a cell has and assing it in the grid
+            neighbour_amount_matrix[row_index][column_index] =
+                get_neighbours_for(row_index, column_index, matrix);
+        }
+    }
+    neighbour_amount_matrix
+}
+//Get neighbours
+fn get_neighbours_for(row_index: usize, col_index: usize, matrix: &[Vec<i32>]) -> i32 {
```

Comment:
I think this function name is not very clear. By looking at the function logic, it seems this function is returning the amount of alive neighbors of a given cell, so an alternative might be:

```suggestion
fn get_alive_neighbours(row_index: usize, col_index: usize, matrix: &[Vec<i32>]) -> i32 {
```

Reply:
Good recomendation, function was renamed